packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -271,22 +297,48 @@ class AndroidStudio implements Comparable<AndroidStudio> {
       });
     }
 
-    // Read all $HOME/.AndroidStudio*/system/.home files. There may be several
-    // pointing to the same installation, so we grab only the latest one.
-    if (globals.fsUtils.homeDirPath != null &&
-        globals.fs.directory(globals.fsUtils.homeDirPath).existsSync()) {
-      final Directory homeDir = globals.fs.directory(globals.fsUtils.homeDirPath);
-      for (final Directory entity in homeDir.listSync(followLinks: false).whereType<Directory>()) {
-        if (!entity.basename.startsWith('.AndroidStudio')) {
-          continue;
-        }
+    // Read all $HOME/.AndroidStudio*/system/.home
+    // or $HOME/.cache/Google/AndroidStudio*/.home files.
+    // There may be several pointing to the same installation,
+    // so we grab only the latest one.
+    final String homeDirPath = globals.fsUtils.homeDirPath;
+
+    if (homeDirPath != null && globals.fs.directory(homeDirPath).existsSync()) {
+      final Directory homeDir = globals.fs.directory(homeDirPath);
+
+      // List of directories to search an installation
```

Comment:
Avoid adding comments that describe what the code is doing exactly, also on L316 + L321